ntbs-service/Services/ValidationService.cs

```diff
@@ -128,8 +128,8 @@ private ContentResult GetValidationResult(object model, IEnumerable<string> keys
                     if (modelStateByKey?.ValidationState == ModelValidationState.Invalid)
                     {
                         errorMessageMap.Add(errorIndex, modelStateByKey.Errors[0].ErrorMessage);
-                        errorIndex++;
                     }
+                    errorIndex++;
```

Comment:
Not sure about this change - now it's just counting keys, not errors?

Reply:
The way this was intended when I first wrote this method is to iterate through all the keys and then associate an error with each, so the index of the error matches the index of the field in the view. If we don't iterate it, and there is no error on one field, the error will get associated with the wrong field in the frontend. That's what happened for DateFrom and DateTo before I changed this (it's also used on clinical details, but there we validate both fields so the error is always shown on both fields).

This method is also used by postcode lookup, but kind of not in the way it was intended (it's validating two fields, postcode and postcodelookup, but the lookup isn't actually a field in the form). I made a change the ValidatePostcode.ts to fix this

Reply:
Ah, ok, gotcha - I feel that the name might be a bit misleading in this case, and specifically the `if (errorIndex > 0)` condition below makes it look like we're trying to count the number of errors that have appeared.
Would it be more telling if we used a for loop in this case?

Reply:
Ah this line has now changed to `if (errorMessageMap.Count > 0)`, so hopefully that should be clear enough :)